src/gConsent/guard/isGConsentAttributes.ts

```diff
@@ -50,3 +60,59 @@ export function isGConsentAttributes(
     isStringArray(x.forPersonalData)
   );
 }
+
+export function isRdfjsGConsentAttributes(
+  dataset: DatasetCore,
+  // eslint-disable-next-line camelcase
+  consent: Quad_Subject,
+) {
+  // isResourceAccessModeArray
+  const modeQuads = dataset.match(consent, acl.mode, null, defaultGraph());
+  if (modeQuads.size === 0) {
+    return false;
+  }
+  for (const { object: mode } of modeQuads) {
+    if (
+      ![acl.Append, acl.Read, acl.Write].some((allowedMode) =>
+        allowedMode.equals(mode),
+      )
+    ) {
+      return false;
+    }
+  }
+
+  // isGConsentStatus
+  const statuses = [
+    ...dataset.match(consent, gc.hasStatus, null, defaultGraph()),
+  ];
+  if (
+    statuses.length !== 1 ||
+    ![
+      gc.ConsentStatusDenied,
+      gc.ConsentStatusExplicitlyGiven,
+      gc.ConsentStatusRequested,
+    ].some((e) => e.equals(statuses[0].object))
+  ) {
+    return false;
+  }
+
+  const forPersonalData = dataset.match(
+    consent,
+    gc.forPersonalData,
+    null,
+    defaultGraph(),
+  );
+
+  // FIXME: Should we really have this condition?
+  if (forPersonalData.size === 0) {
+    throw new Error("No Personal Data specified for Access Grant");
+  }
```

Comment:
FYI @NSeydoux this condition means that this function will error if the access grant has no resources.

Reply:
We used to check if there was an array. I think it was assumed that it was non-empty if one was present; but was didn't explicitly check this.